src/recommendations/RecommendationsPage.jsx

```diff
@@ -29,14 +29,26 @@ const RecommendationsPage = (props) => {
     if (registrationResponse) {
       let coursesWithKeys = [];
       getPersonalizedRecommendations(educationLevel).then((response) => {
-        if (response.length) {
-          coursesWithKeys = response.map(course => ({
-            ...course,
-            courseKey: convertCourseRunKeytoCourseKey(course.activeRunKey),
-          }));
+        coursesWithKeys = response.map(course => ({
+          ...course,
+          courseKey: convertCourseRunKeytoCourseKey(course.activeRunKey),
+        }));
+
+        if (coursesWithKeys.length >= RECOMMENDATIONS_COUNT) {
           setRecommendations(coursesWithKeys.slice(0, RECOMMENDATIONS_COUNT));
+        } else {
+          const courseRecommendations = coursesWithKeys.concat(GENERAL_RECOMMENDATIONS);
+          // Remove duplicate recommendations
+          const uniqueRecommendations = courseRecommendations.filter(
+            (recommendation, index, self) => index === self.findIndex((existingRecommendation) => (
+              existingRecommendation.courseKey === recommendation.courseKey
+            )),
+          );
+          setRecommendations(uniqueRecommendations.slice(0, RECOMMENDATIONS_COUNT));
         }
+
         setIsLoading(false);
+        // We only want to track the recommendations returned by Algolia
```

Comment:
does that mean we will be sending only those recommendations in track event which are returned by algolia?

Reply:
Yes. We want to keep track of what Algolia returned for a user so that we can see if those are the correct recs.

Reply:
Then we should make the course keys array before concatenating the general recs. We need to move it up.